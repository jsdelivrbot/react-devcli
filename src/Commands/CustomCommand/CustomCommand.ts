import {Arguments} from 'yargs';
import ACommand from '../ACommand';
import {ILintable, IMakeable, IRemovable} from '../CommandContracts';
import {CommandsConfig} from 'react-devcli';
import {Command} from '../CommandDecorator';
import {description, flags, name} from './CustomCommandOptions';

@Command(name, description, flags)
class CustomCommand extends ACommand implements IRemovable, IMakeable, ILintable {
    public run(argv: Arguments, config: CommandsConfig): void {
        console.log(config);
        console.log(argv);
    }

    public help(argv: Arguments): void {
        console.log(argv);
    }

    public remove(argv: Arguments): void {
        console.log(argv);
    }

    public make(argv: Arguments): void {
        console.log(argv);
    }

    public lint(argv: Arguments): void {
        console.log(argv);
    }
}

export default CustomCommand;
