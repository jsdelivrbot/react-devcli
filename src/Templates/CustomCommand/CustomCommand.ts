export const js = (): string => {
    return '';
};

export const ts = (): string => {
    return '';
};
