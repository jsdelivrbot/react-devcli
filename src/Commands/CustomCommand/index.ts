import CustomCommand from './CustomCommand';

export default new CustomCommand();
