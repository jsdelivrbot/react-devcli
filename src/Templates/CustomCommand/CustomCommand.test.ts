import {js, ts} from './CustomCommand';

describe('CustomCommand template test suite', (): void => {
    test('should match custom command template snapshot for JS language', (): void => {
        expect(js()).toMatchSnapshot();
    });

    test('should match custom command template snapshot for TS language', (): void => {
        expect(ts()).toMatchSnapshot();
    });
});
