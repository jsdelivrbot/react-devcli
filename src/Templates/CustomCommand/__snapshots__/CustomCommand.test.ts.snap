// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCommand template test suite should match custom command template snapshot for JS language 1`] = `""`;

exports[`CustomCommand template test suite should match custom command template snapshot for TS language 1`] = `""`;
